examples/knative/powercli/kn-pcli-ha-restarted-vms/handler.ps1

```diff
@@ -0,0 +1,197 @@
+   $VCENTER_PASSWORD = ${jsonSecrets}.VCENTER_PASSWORD
+   $VCENTER_CERTIFICATE_ACTION = ${jsonSecrets}.VCENTER_CERTIFICATE_ACTION
+
+   # Configure TLS 1.2/1.3 support as this is required for latest vSphere release
+   [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12 -bor [System.Net.SecurityProtocolType]::Tls13
+
+   Write-Host "$(Get-Date) - Configuring PowerCLI Configuration Settings`n"
+   Set-PowerCLIConfiguration -InvalidCertificateAction:${VCENTER_CERTIFICATE_ACTION} -ParticipateInCeip:$true -Confirm:$false
+
+   Write-Host "$(Get-Date) - Connecting to vCenter Server $VCENTER_SERVER`n"
+
+   try {
+      Connect-VIServer -Server $VCENTER_SERVER -User $VCENTER_USERNAME -Password $VCENTER_PASSWORD
+   }
+   catch {
+      Write-Error "$(Get-Date) - ERROR: Failed to connect to vCenter Server"
+      throw $_
+   }
+
+   Write-Host "$(Get-Date) - Successfully connected to $VCENTER_SERVER`n"
+
+   Write-Host "$(Get-Date) - Init Processing Completed`n"
+}
+
+Function Process-Shutdown {
+   [CmdletBinding()]
+   param()
+   Write-Host "$(Get-Date) - Processing Shutdown`n"
+
+   Write-Host "$(Get-Date) - Disconnecting from vCenter Server`n"
+
+   try {
+      Disconnect-VIServer * -Confirm:$false
+   }
+   catch {
+      Write-Error "$(Get-Date) - Error: Failed to Disconnect from vCenter Server"
+   }
+
+   Write-Host "$(Get-Date) - Shutdown Processing Completed`n"
+}
+
+Function Process-Handler {
+   [CmdletBinding()]
+   param(
+      [Parameter(Position=0,Mandatory=$true)][CloudNative.CloudEvents.CloudEvent]$CloudEvent
+   )
+
+   # Decode CloudEvent
+   try {
+      $cloudEventData = $cloudEvent | Read-CloudEventJsonData -Depth 10
+   }
+   catch {
+      throw "`nPayload must be JSON encoded"
+   }
+
+   try {
+      $jsonSecrets = ${env:HA_SECRET} | ConvertFrom-Json
+   }
+   catch {
+      throw "`nK8s secret `$env:HA_SECRET does not look to be defined"
+   }
+
+   # Main processing of gathering specific events
+   $report = @()
+   $eventnumber = 1000
+   # get vCenter EventManager
+   try {
+      $si = get-view ServiceInstance
+   }
+   catch {
+      Write-Host "$(Get-Date) - ERROR: Could not retrieve Service Instance"
+      throw $_
+   }
+
+   try {
+      $em = get-view $si.Content.EventManager
+   }
+   catch {
+      Write-Host "$(Get-Date) - ERROR: Could not retrieve Event Manager"
+      throw $_
+   }
+
+   # Create Event Filter Spec
+   $EventFilterSpec = New-Object VMware.Vim.EventFilterSpec
+   # Get specific Events based on EventTypeIDs for EventFilterSpec
+   $tgtEvtTypeIDs = "com.vmware.vc.ha.VmRestartedByHAEvent", "com.vmware.vc.HA.DasHostFailedEvent"
+   $EventFilterSpec.EventTypeId = $tgtEvtTypeIDs
+   # Time range to look for specific IDs (current day specified) for EventFilterSpec
+   $EventFilterSpecByTime = New-Object VMware.Vim.EventFilterSpecByTime
+   $EventFilterSpecByTime.BeginTime = [datetime]::Today
```

Comment:
question: is Today set to today 0AM UTC?

Reply:
It would be 00:00 local time, but yes

> [datetime]::Today

Tuesday, April 26, 2022 12:00:00 AM